### PR TITLE
GH-12335 Refactor app/session.go to use structured logging

### PR DESCRIPTION
**Pull request by lurcio**
#### Summary
Refactor "app/session.go" to use structured logging.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12335

### CODE REVIEW

File: app/session.go [hunk 1]
```diff
@@ -96,7 +95,7 @@ func (a *App) GetSessions(userId string) ([]*model.Session, *model.AppError) {
 func (a *App) UpdateSessionsIsGuest(userId string, isGuest bool) {
 	sessions, err := a.Srv.Store.Session().GetSessions(userId)
 	if err != nil {
-		mlog.Error(fmt.Sprintf("Unable to get user sessions: userId=%s err=%s", userId, err.Error()))
+		mlog.Error("Unable to get user sessions: userId=%s err=%s", mlog.String("user_id", userId), mlog.Err(err))
```
Comment (reflog) [inline app/session.go]: When using structured logging, %s is not needed. This line should be: `mlog.Error("Unable to get user sessions", mlog.String("user_id", userId), mlog.Err(err))`

File: app/session.go [hunk 2]
```diff
@@ -107,7 +106,7 @@ func (a *App) UpdateSessionsIsGuest(userId string, isGuest bool) {
 		}
 		err := a.Srv.Store.Session().UpdateProps(session)
 		if err != nil {
-			mlog.Error(fmt.Sprintf("Unable to update isGuest session: %s", err.Error()))
+			mlog.Error("Unable to update isGuest session: %s", mlog.Err(err))
```
Comment (reflog) [inline app/session.go]: same as above

File: app/session.go [hunk 3]
```diff
@@ -214,7 +213,7 @@ func (a *App) RevokeSessionsForDeviceId(userId string, deviceId string, currentS
 	}
 	for _, session := range sessions {
 		if session.DeviceId == deviceId && session.Id != currentSessionId {
-			mlog.Debug(fmt.Sprintf("Revoking sessionId=%v for userId=%v re-login with same device Id", session.Id, userId), mlog.String("user_id", userId))
+			mlog.Debug("Revoking sessionId=%v for userId=%v re-login with same device Id", mlog.String("session_id", session.Id), mlog.String("user_id", userId))
```
Comment (reflog) [inline app/session.go]: same as above

File: app/session.go [hunk 4]
```diff
@@ -279,7 +278,7 @@ func (a *App) UpdateLastActivityAtIfNeeded(session model.Session) {
 	}
 
 	if err := a.Srv.Store.Session().UpdateLastActivityAt(session.Id, now); err != nil {
-		mlog.Error(fmt.Sprintf("Failed to update LastActivityAt for user_id=%v and session_id=%v, err=%v", session.UserId, session.Id, err), mlog.String("user_id", session.UserId))
+		mlog.Error("Failed to update LastActivityAt for user_id=%v and session_id=%v, err=%v", mlog.String("user_id", session.UserId), mlog.String("session_id", session.Id), mlog.Err(err))
```
Comment (reflog) [inline app/session.go]: same as above

File: app/session.go (hunk 1 above)
Comment (lurcio) [reply app/session.go]: Thank you for the feedback @reflog. I have made some changes. I reworded one of the error messages slightly to hopefully make more sense.

**Comment by reflog** [comment]
@lurcio - please sync the branch with master and we'll be able to merge!

**Comment by lurcio** [comment]
@reflog done, but tests break now 😕

**Comment by reflog** [comment]
@lurcio - it was a flaky test. Restarted - all is good!

**Pull request merged**